lib/errors/error-collector.js

```diff
@@ -89,69 +89,47 @@ class ErrorCollector {
   }
 
   /**
-   * Helper method for processing errors that are created with .noticeError()
+   * Gets the iterable property from the transaction based on the error type
+   * @param {Transaction} transaction the collected exception's transaction
+   * @param {string} errorType the type of error to be processed: "user", "transactionException", "transaction"
+   * @returns
+   */
```

Comment:
In my opinion, it is easier to read, and maintain, a JSDoc block when a little extra formatting is applied. As an example, I prefer to write them like:

```js
/**
 * A helpful introduction, hard wrapped at 80 characters.
 *
 * @param {object} params Input parameters.
 * @param {boolean} params.hasSomething Indicates if something is present.
 * @param {string} params.type The type of something. Possible values
 * are: "foo", "bar", or "baz".
 * @param {object} params.something The something to work with.
 *
 * @throws {Error} When something bad happens.
 * @returns {object} A new something!
 */
function changeSomething(params) {
  // do stuff
}
```

Notice the breaks between the introduction, the parameter definitions, and the results sections.